app/actions/remote/post.ts

```diff
@@ -552,3 +552,36 @@ export const togglePinPost = async (serverUrl: string, postId: string) => {
         return {error};
     }
 };
+
+export const markPostAsUnread = async (serverUrl: string, postId: string, channelId: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+    try {
+        const userId = await queryCurrentUserId(database);
+        const myChannel = await queryMyChannel(database, channelId);
+
+        if (myChannel) {
```

Comment:
@anurag6713  FYI.  Just double-checking, you might have to update this method to mark thread as unread